main.cpp

```diff
@@ -120,17 +127,42 @@ int main()
     /*
      construct 2 more Car instances via emplace_back.
      */
+    cars.emplace_back("patrice");
+    cars.emplace_back("fabrice");
     
     /*
      now reserve and emplace_back your Trucks and Motorcyles
      */
+    motorcycles.reserve(3);
+    motorcycles.emplace_back("rocknroll");
+    motorcycles.emplace_back("scootteam");
+    motorcycles.emplace_back("fastarrow");
     
+    trucks.reserve(3);
+    trucks.emplace_back("alizon");
+    trucks.emplace_back("amashop");
+    trucks.emplace_back("quasibaba");
     
     
-    
-    assert(false);
+    //assert(false);
     //add the cars, motorcycles and trucks to the highway using range-based for() loops: for( element : vec ) { ... }
     //be careful to not accidentally make element copies when iterating.
+
+    for(auto& car : cars)
+    {
+        highway.addVehicle(&car); 
+
+        for(auto& motorcycle : motorcycles)
+        {
+            highway.addVehicle(&motorcycle); 
+
+            for(auto& semitruck : trucks)
+            {
+                highway.addVehicle(&semitruck); 
+            }
+        }
+    }
```

Comment:
Do not use nested loops.
use one loop for each container type
For each motorcycle, you add 3 semitrucks.  You have 3 motorcycles, so that's 9 semitrucks that get added.
Then you do the same thing for the cars
for each car, you add 3 motorcycles, and for each motorcycle you add 3 semitrucks.
Do not do this.

Add the cars and only the cars in one loop
Add the motorcycles and only the motorcycles in a different loop
Add the trucks and only the trucks in a different loop.
- [ ] revise